cmd/gcContent/gcContent.go

```diff
@@ -0,0 +1,96 @@
+// Command Group: "FASTA and Multi-FASTA Tools"
+
+// Calculates the gc content of fasta sequences for all regions specified in a bed file
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fasta"
+	"github.com/vertgenlab/gonomics/fileio"
+	"log"
+)
+
+func gcContent(bedFile string, faFile string, outFile string, multiFaMode bool, species string) {
+
+	var currRecordSeq []dna.Base
+	var found bool
+	var gc float64
+	var annotated bed.Bed
+	var speciesStart, speciesEnd int
+
+	regionsChan := bed.GoReadToChan(bedFile)
+	records := fasta.Read(faFile)
+	recordsMap := fasta.ToMap(records)
+	out := fileio.EasyCreate(outFile)
+
+	for curr := range regionsChan {
+
+		if multiFaMode {
+			// when multiFa, each fasta sequence is a species, e.g. >human, >hca, assuming that the user specified the correct chromosome, e.g. this multi-fasta is chr1.fa
+			currRecordSeq, found = recordsMap[species]
+			if found {
+				annotated = bed.Bed{Chrom: curr.Chrom, ChromStart: curr.ChromStart, ChromEnd: curr.ChromEnd, FieldsInitialized: 4}
+				// need to convert bed region (reference sequence e.g. human RefPos), to AlnPos, then use AlnPos to index requested sequence (e.g. hca)
+				speciesStart = fasta.RefPosToAlnPos(records[0], curr.ChromStart)
+				speciesEnd = fasta.RefPosToAlnPos(records[0], curr.ChromEnd)
+				gc = dna.GCContent(currRecordSeq[speciesStart:speciesEnd])
+				annotated.Name = fmt.Sprintf("%e", gc)
```

Comment:
I don't really have an issue with it, but what is the thinking behind outputting the GC percent in scientific notation?

Reply:
No specific reason, just a way to output a float number